drivers/pinctrl/pinctrl_max32.c

```diff
@@ -67,20 +66,7 @@ static int pinctrl_configure_pin(pinctrl_soc_pin_t soc_pin)
 		gpio_cfg.vssel = MXC_GPIO_VSSEL_VDDIO;
 	}
 
-	switch (pincfg & MAX32_GPIO_DRV_STRENGTH_MASK) {
-	case MAX32_GPIO_DRV_STRENGTH_1:
-		gpio_cfg.drvstr = MXC_GPIO_DRVSTR_1;
-		break;
-	case MAX32_GPIO_DRV_STRENGTH_2:
-		gpio_cfg.drvstr = MXC_GPIO_DRVSTR_2;
-		break;
-	case MAX32_GPIO_DRV_STRENGTH_3:
-		gpio_cfg.drvstr = MXC_GPIO_DRVSTR_3;
-		break;
-	default:
-		gpio_cfg.drvstr = MXC_GPIO_DRVSTR_0;
-		break;
-	}
+	gpio_cfg.drvstr = (pincfg >> MAX32_DRV_STRENGTH_SHIFT) & (0x3);
```

Comment:
You could use `FIELD_GET`

Reply:
Hi Muareen thanks for suggestion. I checked this but there's a division there that i think didn't yield to same results
#define FIELD_GET(mask, value)  (((value) & (mask)) / LSB_GET(mask))

for example FIELD_GET(0x2, 0x777) gives 1, but in my case it should've given 2.

or maybe i'm missing something.